multichain-testing/test/basic-flows.test.ts

```diff
@@ -22,13 +22,21 @@ test.before(async t => {
   t.context = { ...rest, wallets, deleteTestKeys };
 
   t.log('bundle and install contract', contractName);
-  await t.context.deployBuilder(contractBuilder);
-  const { vstorageClient } = t.context;
-  await t.context.retryUntilCondition(
-    () => vstorageClient.queryData(`published.agoricNames.instance`),
-    res => contractName in Object.fromEntries(res),
-    `${contractName} instance is available`,
+  const { vstorageClient, deployBuilder, retryUntilCondition } = t.context;
+  const installedContracts = await vstorageClient.queryData(
+    `published.agoricNames.instance`,
   );
+  const isInstalled = contractName in Object.fromEntries(installedContracts);
+  if (!isInstalled) {
```

Comment:
reads well but a little surprising to see the map made one-off. also these are instances, not installations.
```suggestion
  const instances = Object.fromEntries(await vstorageClient.queryData(
    `published.agoricNames.instance`,
  ));
  if (!contractName in instances) {
```